android-core/src/main/java/com/mparticle/networking/MParticleBaseClientImpl.java

```diff
@@ -92,14 +95,20 @@ public long getNextRequestTime(Endpoint endpoint) {
     }
 
     protected MPUrl getUrl(Endpoint endpoint) throws MalformedURLException {
-        return getUrl(endpoint, null);
+        return getUrl(endpoint, null,null);
     }
 
-    protected MPUrl getUrl(Endpoint endpoint, @Nullable String identityPath) throws MalformedURLException {
-        return getUrl(endpoint, identityPath, false);
+    protected MPUrl getUrl(Endpoint endpoint, @Nullable long mpId) throws MalformedURLException {
+        HashMap<String, String> queryParams = new HashMap<>();
+        queryParams.put("mpid", String.valueOf(mpId));
+        return getUrl(endpoint, null, queryParams);
     }
 
-    protected MPUrl getUrl(Endpoint endpoint, @Nullable String identityPath, boolean forceDefaultUrl) throws MalformedURLException {
+    protected MPUrl getUrl(Endpoint endpoint, @Nullable String identityPath, HashMap<String, String> queryParams) throws MalformedURLException {
+        return getUrl(endpoint, identityPath, false,null);
+    }
+
+    protected MPUrl getUrl(Endpoint endpoint, @Nullable String identityPath, boolean forceDefaultUrl, HashMap<String, String> queryParams) throws MalformedURLException {
```

Comment:
```suggestion
    protected MPUrl getUrl(Endpoint endpoint, @Nullable String identityPath, boolean forceDefaultUrl, HashMap<String, String> audienceQueryParams) throws MalformedURLException {
```

we should either:
1. update the implementation to always append these query parameters if not null, or
2. name this parameter audienceQueryParams so that the caller knows they are only used in audience.